specification/cognitiveservices/data-plane/ContentSafety/stable/2024-09-01/contentsafety.json

```diff
@@ -0,0 +1,1304 @@
+          "format": "byte",
+          "description": "The Base64 encoding of the image."
+        },
+        "blobUrl": {
+          "type": "string",
+          "format": "uri",
+          "description": "The blob url of the image."
+        }
+      }
+    },
+    "PagedTextBlocklist": {
+      "type": "object",
+      "description": "Paged collection of TextBlocklist items",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The TextBlocklist items on this page",
+          "items": {
+            "$ref": "#/definitions/TextBlocklist"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "PagedTextBlocklistItem": {
+      "type": "object",
+      "description": "Paged collection of TextBlocklistItem items",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The TextBlocklistItem items on this page",
+          "items": {
+            "$ref": "#/definitions/TextBlocklistItem"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "PromptInjectionAnalysisResult": {
+      "type": "object",
+      "description": "The individual analysis result of potential direct or indirect injection attacks.",
+      "properties": {
+        "attackDetected": {
+          "type": "boolean",
+          "description": "Whether a potential injection attack is detected or not."
+        }
+      },
+      "required": [
+        "attackDetected"
+      ]
+    },
+    "RemoveTextBlocklistItemsOptions": {
+      "type": "object",
+      "description": "The request to remove blocklistItems from a text blocklist.",
+      "properties": {
+        "blocklistItemIds": {
+          "type": "array",
+          "description": "Array of blocklistItemIds to remove.",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "required": [
+        "blocklistItemIds"
+      ]
+    },
+    "ShieldPromptOptions": {
+      "type": "object",
+      "description": "The request of analyzing potential direct or indirect injection attacks.",
+      "properties": {
+        "userPrompt": {
+          "type": "string",
+          "description": "The user prompt needs to be analyzed, which may contain direct injection attacks.",
+          "maxLength": 10000
+        },
+        "documents": {
+          "type": "array",
+          "description": "The documents needs to be analyzed, which may contain direct or indirect injection attacks.",
+          "items": {
+            "type": "string"
```

Comment:
Is there a maxLength for the documents?

Reply:
My question was actually about the maxLength of individual documents -- not the number of elements in the array.  You previously had `maxLength: 10000` on `userPrompt` so I was suggesting something similar here. But I see that you have removed that. I'm not a fan of omitting information from a REST API definition so that it can be "silently" changed later, but that's maybe a discussion we can put off.